components/pedigree/ui/src/main/resources/PhenoTips/PedigreeImageService.xml

```diff
@@ -37,13 +37,36 @@
   <minorEdit>false</minorEdit>
   <syntaxId>xwiki/2.1</syntaxId>
   <hidden>true</hidden>
-  <content>{{include reference="PhenoTips.PedigreeMacros" /}}
-
+  <content>
 {{velocity}}
+#if ("$!request.action" == 'export')
+  {{html wiki="false" clean="false"}}
+  $response.setContentType('application/json')##
+  #if ("$!request.image" == '')
+    $response.setStatus(404)##
+    {"error" : "$services.localization.render('phenotips.PedigreeImageService.noPedigreeImage')"}
+    #break
+  #end
+  ## Fix broken SVG generated by a broken version of the pedigree editor
+  #if ($stringtool.countMatches($request.image, 'xmlns=') &gt; 1)
```

Comment:
There should be a reason @sdumitriu placed those lines in this image service XML file and not in workspace.js in this commit https://github.com/phenotips/phenotips/commit/0c74fa5c69bad684606ce6da9048b6575e8f0896
Especially if two days later there was your commit to workspace.js with line doing pretty much same with replacing xmlns-ses to fix generating SVG in IE9 https://github.com/phenotips/phenotips/commit/736ec819ffa93d4fceb4c4538e342cf1d2b7204f
From the header of @sdumitriu 's commit "PT-2249: The pedigree image service fails to generate PNGs for pedigrees created with 1.2.1" and the comment in commit code " Fix broken SVG generated by a broken version of the pedigree editor" am guessing that those fixes are placed in image service XML and not in pedigree code for the reason of fixing issues if someone will install old broken version of pedigree editor 1.2.1and placing fixes in newer pedigree code wont help in this case. Just a thought..